db_sql.go

```diff
@@ -469,89 +539,155 @@ func SqlCreateDatabase(conx *DBConnection) error {
+		CREATE UNIQUE INDEX idx_authuser_identity ON authuser (identity);`,
+
+		`DROP INDEX idx_authuser_identity;
+		CREATE UNIQUE INDEX idx_authuser_identity ON authuser (LOWER(identity));`,
+
+		`ALTER TABLE authuser ADD COLUMN pwdtoken VARCHAR;`,
+
+		//Change from using identity as unique reference to user account to using userid
+		`CREATE TABLE userstore (userid BIGSERIAL PRIMARY KEY, identity VARCHAR);
+		CREATE UNIQUE INDEX idx_userstore_identity ON userstore (LOWER(identity));
+		INSERT INTO userstore (identity) SELECT identity from authuser;
+
+		CREATE TABLE authsession_temp(id BIGSERIAL PRIMARY KEY, sessionkey VARCHAR, identity VARCHAR, permit VARCHAR, expires TIMESTAMP);
+		INSERT INTO authsession_temp (id, sessionkey, identity, permit, expires) SELECT * FROM authsession;
+		TRUNCATE authsession;
+		DROP INDEX idx_authsession_identity;
+		ALTER TABLE authsession DROP COLUMN identity;
+		ALTER TABLE authsession ADD COLUMN userid BIGSERIAL;
+		CREATE INDEX idx_authsession_userid ON authsession (userid);
+		INSERT INTO authsession (id, sessionkey, userid, permit, expires)
+		SELECT authtemp.id, authtemp.sessionkey, us.userid, authtemp.permit, authtemp.expires FROM authsession_temp AS authtemp
+		JOIN userstore AS us ON us.identity = authtemp.identity;
+		DROP TABLE IF EXISTS authsession_temp;
+
+		CREATE TABLE authuser_temp(id BIGSERIAL PRIMARY KEY, identity VARCHAR, password VARCHAR, permit VARCHAR, pwdtoken VARCHAR);
+		INSERT INTO authuser_temp (id, identity, password, permit, pwdtoken) SELECT * FROM authuser;
+		TRUNCATE authuser;
+		DROP INDEX idx_authuser_identity;
+		ALTER TABLE authuser DROP COLUMN identity;
+		ALTER TABLE authuser ADD COLUMN userid BIGSERIAL;
+		ALTER TABLE authuser ADD CONSTRAINT fk_storeUserId_authUserId FOREIGN KEY (userid) REFERENCES userstore (userid);
+		INSERT INTO authuser (id, userid, password, permit, pwdtoken)
+		SELECT usertemp.id, us.userid, usertemp.password, usertemp.permit, usertemp.pwdtoken FROM authuser_temp AS usertemp
+		JOIN userstore AS us ON us.identity = usertemp.identity;
+		DROP TABLE IF EXISTS authuser_temp;`,
+	}
+
+	for _, src := range text {
+		srcCapture := src
+		migrations = append(migrations, func(tx migration.LimitedTx) error {
+			_, err := tx.Exec(srcCapture)
+			return err
+		})
+	}
+	return migrations
+}
+
+//This function makes sure that the old migrations that were part of the old migration tool have run.
+func runBootstrap(db *sql.DB) (bool, error) {
+	oldVersionTables := []string{"authgroup_version", "authsession_version", "authuser_version"}
+	// We first want to see if the new migrator has taken over, by checking if old version tracking tables exist
+	tableNameResult := ""
+	query := db.QueryRow(fmt.Sprintf("SELECT table_name FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_NAME = '%v'", oldVersionTables[0]))
+	err := query.Scan(&tableNameResult)
+	if err != nil {
+		if strings.Index(err.Error(), "no rows in result set") != -1 {
```

Comment:
I'm skeptical on there being two "up to date" return paths. Under which conditions will one hit the first `return true, nil` branch, and under which conditions will one hit the second such branch?